subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -426,4 +431,74 @@ public void applyTo(StartParameterInternal settings, Origin origin) {
             settings.setExportKeys(true);
         }
     }
+
+    public static class ConfigurationCacheOption extends EnumBuildOption<ConfigurationCacheOption.Value, StartParameterInternal> {
+
+        public enum Value {
+            OFF, ON, WARN
+        }
+
+        public static final String PROPERTY_NAME = "org.gradle.unsafe.configuration-cache";
+        public static final String LONG_OPTION = "configuration-cache";
+
+        public ConfigurationCacheOption() {
+            super(
+                LONG_OPTION,
+                Value.class,
+                Value.values(),
+                PROPERTY_NAME,
+                CommandLineOptionConfiguration.create(
+                    LONG_OPTION,
+                    "Enables the configuration cache (off, on, or warn). Gradle will try to reuse the build configuration from previous builds."
```

Comment:
Maybe add [incubating] to the description.

Reply:
It's already there thanks to the `.incubating()` call below.